contracts/middle-layer/resource-mirror/MultiMessage.sol

```diff
@@ -0,0 +1,125 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+
+pragma solidity ^0.8.0;
+
+import "@openzeppelin/contracts-upgradeable/proxy/utils/Initializable.sol";
+import "../../interface/ICrossChain.sol";
+import "../../interface/IMultiMessage.sol";
+import "./CmnHub.sol";
+import "./storage/MultiStorage.sol";
+
+contract MultiMessage is MultiStorage, CmnHub, IMultiMessage {
+    uint8 public constant ACK_PACKAGE = 0x01;
+    uint8 public constant FAIL_ACK_PACKAGE = 0x02;
+
+    mapping(address => bool) public whitelistTargets;
+
+    constructor() {
+        _disableInitializers();
+    }
+
+    /*----------------- initializer -----------------*/
+    function initialize(address _ERC721_token, address _additional) public initializer {
+        __cmn_hub_init_unchained(_ERC721_token, _additional);
+
+        channelId = MULTI_MESSAGE_CHANNEL_ID;
+
+        whitelistTargets[BUCKET_HUB] = true;
+        whitelistTargets[GROUP_HUB] = true;
+        whitelistTargets[OBJECT_HUB] = true;
+        whitelistTargets[TOKEN_HUB] = true;
+    }
+
+    function initializeV2() public reinitializer(2) {
```

Comment:
No need for this. Just add `__cmn_hub_init_unchained_v2(INIT_MAX_CALLBACK_DATA_LENGTH);` to `initialize` function.

Reply:
This is to keep consistent with other middle-layer contracts in terms of initialization form.
The advantage of this is that there is no need for additional effort from deployers/reviewers to get points from those contracts requiring `initializeV2`, which are not required